config/OpenAPI/templates.php

```diff
@@ -0,0 +1,167 @@
+<?php
+return [
+    /**
+     * OpenAPI templates
+     *
+     * Each item in `paths` and `components` contains a `{$resource}` placeholder
+     * that will be replaced with actual resource and object types names
+     *
+     * Note:
+     *  - `{$resource}_attributes`, `{$resource}_meta` and `{$resource}_relationships` will be dynamically created
+     *
+     */
+    'OATemplates' => [
+        'paths' => [
+            '/{$resource}/{id}' => [
```

Comment:
how can we deal with special endpoints like `admin` and `model`, where `$resource` does not match exactly endpoint path? for instance: `/admin/:op/:id`. Would we accept `$resource` = `'admin/:op'`, or should we use some other pattern to handle these special endpoints?
however, it's fine for now, as long as we are dealing with `roles` and `streams`

Reply:
You are right, we could introduce a ```{$prefix}``` and have something like ```/{$prefix}{$resource}/{id}``` or use separate files like ```admin_endpoint.php``` without templates.

It depends on how consistent the API structure will be, hopefully a lot :wink: